modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/UpgradeSetResultsSetTotalMethodCheck.java

```diff
@@ -5,78 +5,167 @@
+		return new String[] {"java", "jsp", "jspf"};
+	}
+
+	private String _formatJava(String content, String fileName)
+		throws Exception {
 
+		JavaClass javaClass = JavaClassParser.parseJavaClass(fileName, content);
+
+		if (javaClass == null) {
 			return content;
 		}
 
-		content = _removeSetTotal(content);
-		content = _replaceSetResults(content);
+		for (JavaTerm childJavaTerm : javaClass.getChildJavaTerms()) {
+			if (!childJavaTerm.isJavaMethod()) {
+				continue;
+			}
+
+			JavaMethod javaMethod = (JavaMethod)childJavaTerm;
+
+			String javaMethodContent = javaMethod.getContent();
+
+			content = _getNewContent(javaMethodContent, content);
+		}
 
 		return content;
 	}
 
-	private String _removeSetTotal(String content) {
-		String newContent = content;
+	private String _getNewContent(String content, String fileContent) {
+		String newContent = fileContent;
 
-		Matcher setTotalMatcher = _setTotalPattern.matcher(content);
+		Matcher matcher = _setResultsPattern.matcher(content);
 
-		while (setTotalMatcher.find()) {
-			if (hasClassOrVariableName(
-					"SearchContainer", content, content,
-					JavaSourceUtil.getMethodCall(
-						content, setTotalMatcher.start()))) {
+		String setResultMethodCall = null;
 
-				newContent = StringUtil.removeSubstring(
-					newContent, setTotalMatcher.group());
-			}
+		int positionSetResult = 0;
+
+		if (matcher.find()) {
+			positionSetResult = matcher.start();
+
+			setResultMethodCall = JavaSourceUtil.getMethodCall(
+				content, positionSetResult);
 		}
 
-		return newContent;
+		matcher = _setTotalPattern.matcher(content);
+
+		String setTotalMethodCall = null;
+
+		int positionSetTotal = 0;
+
+		if (matcher.find()) {
+			positionSetTotal = matcher.start();
+
+			setTotalMethodCall = JavaSourceUtil.getMethodCall(
+				content, positionSetTotal);
+		}
+
+		if (!hasClassOrVariableName(
+				"SearchContainer", newContent, newContent,
+				setResultMethodCall) &&
+			!hasClassOrVariableName(
+				"SearchContainer", newContent, newContent,
+				setTotalMethodCall)) {
+
+			return newContent;
+		}
+
+		String firstMethod =
+			(positionSetResult < positionSetTotal) ? setResultMethodCall :
+				setTotalMethodCall;
+
+		String newJavaMethodContent = content;
+
+		if ((setResultMethodCall != null) && (setTotalMethodCall != null)) {
+			newJavaMethodContent = StringUtil.removeSubstring(
+				content,
+				firstMethod + StringPool.SEMICOLON + StringPool.NEW_LINE);
+		}
+
+		return StringUtil.replace(
+			newContent, content,
+			_getNewMethod(
+				firstMethod, newJavaMethodContent, setResultMethodCall,
+				setTotalMethodCall));
 	}
 
-	private String _replaceSetResults(String content) {
-		String newContent = content;
+	private String _getNewMethod(
+		String firstMethod, String javaMethodContent,
+		String setResultMethodCall, String setTotalMethodCall) {
 
-		Matcher setResultsMatcher = _setResultsPattern.matcher(content);
+		String methodCall = setTotalMethodCall;
```

Comment:
@TamyrisTorres, I believe this variable isn't really necessary, how about we do the following instead?

```
		StringBuilder sb = new StringBuilder();

		sb.append(getVariableName(firstMethodCall));
		sb.append(".setResultsAndTotal(");
		sb.append(StringUtil.merge(parameters, StringPool.COMMA_AND_SPACE));
		sb.append(StringPool.CLOSE_PARENTHESIS);

		return StringUtil.replace(javaMethodContent, firstMethodCall, sb.toString());
```

Reply:
@natocesarrego The firstMethodCall variable is the method that will be excluded from the content file. I use firstMethodCall to identify who was deleted.